code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -245,7 +245,7 @@
  * * given_layer - the piping_layer we are connecting to
  */
 /obj/machinery/atmospherics/proc/check_connectable_layer(obj/machinery/atmospherics/target, given_layer)
-	if(target.piping_layer == given_layer || (target.pipe_flags | pipe_flags) & PIPING_ALL_LAYER)
+	if(target.piping_layer == given_layer || target.pipe_flags & PIPING_ALL_LAYER)
```

Comment:
why remove the | here?

Reply:
That's what whas causing the issue, this check Is done by both sides, so when a pipe was connected to the adapter, it filled up the front/back nodes with itself.
It wasn't an issue with other pipes because only those had the PIPING_ALL_LAYERS flag.

Reply:
lmao